VSDiagnostics/VSDiagnostics/VSDiagnostics/Diagnostics/Strings/StringDotFormatWithDifferentAmountOfArguments/StringDotFormatWithDifferentAmountOfArgumentsAnalyzer.cs

```diff
@@ -0,0 +1,150 @@
+            {
+                return;
+            }
+
+            var hasObjectArray = 
+                methodSymbol.Parameters.FirstOrDefault(x => x.Type.Kind == SymbolKind.ArrayType && 
+                                                            ((IArrayTypeSymbol) x.Type).ElementType.SpecialType == SpecialType.System_Object) != null;
+            var hasObject = 
+                methodSymbol.Parameters.FirstOrDefault(x => x.Type.SpecialType == SpecialType.System_Object) != null;
+
+            if (!(hasObject || hasObjectArray))
+            {
+                return;
+            }
+
+            var formatIndex = formatParam.Ordinal;
+            var formatExpression = invocation.ArgumentList.Arguments[formatIndex];
+            var formatString = context.SemanticModel.GetConstantValue(formatExpression.Expression);
+            if (!formatString.HasValue)
+            {
+                return;
+            }
+
+            // We ignore interpolated strings for now (workitem tracked in https://github.com/Vannevelj/VSDiagnostics/issues/313)
+            if (invocation.ArgumentList.Arguments[formatIndex].Expression is InterpolatedStringExpressionSyntax)
+            {
+                return;
+            }
+
+            // Get the total amount of arguments passed in for the format
+            // If the first one is the literal (aka: the format specified) then every other argument is an argument to the format
+            // If not, it means the first one is the CultureInfo, the second is the format and all others are format arguments
+            // We also have to check whether or not the arguments are passed in through an explicit array or whether they use the params syntax
+            var formatArguments = invocation.ArgumentList.Arguments.Skip(formatIndex + 1).ToArray();
+            var amountOfFormatArguments = formatArguments.Length;
+
+            if (formatArguments.Length == 1)
+            {
+                // Inline array creation à la string.Format("{0}", new object[] { "test" })
+                var arrayCreation = formatArguments[0].Expression as ArrayCreationExpressionSyntax;
+                if (arrayCreation?.Initializer?.Expressions != null)
+                {
+                    amountOfFormatArguments = arrayCreation.Initializer.Expressions.Count;
+                }
+
+                // We don't handle method calls
+                var invocationExpression = formatArguments[0].Expression as InvocationExpressionSyntax;
+                if (invocationExpression != null)
+                {
+                    return;
+                }
+
+                // If it's an identifier, we don't handle those that provide an array as a single argument
+                // Other types are fine though -- think about string.Format("{0}", name);
+                var referencedIdentifier = formatArguments[0].Expression as IdentifierNameSyntax;
+                if (referencedIdentifier != null)
```

Comment:
Actually, you should only run this check if the corresponding parameter is `object[]`.

Passing an array as an `object` is fine, and should be treated as a single value.  (although it won't be very useful, since arrays don't override `ToString()`)